app/helpers/ExerciseConfigChecker.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace App\Helpers;
+
+use App\Exceptions\ExerciseConfigException;
+use App\Helpers\ExerciseConfig;
+use App\Helpers\ExerciseConfig\Compilation\CompilationParams;
+use App\Model\Entity\Exercise;
+use App\Model\Entity\RuntimeEnvironment;
+use Nette\SmartObject;
+
+class ExerciseConfigChecker {
+  use SmartObject;
+
+  private $compiler;
+
+  private $validator;
+
+  private $loader;
+
+  public function  __construct(ExerciseConfig\Compiler $compiler, ExerciseConfig\Validator $validator, ExerciseConfig\Loader $loader) {
+    $this->compiler = $compiler;
+    $this->validator = $validator;
+    $this->loader = $loader;
+  }
+
+  /**
+   * Make up names of submitted files for a runtime environment.
+   * This is necessary because we do not have any real submissions yet, but the compiler needs their names.
+   * TODO when we implement a mechanism that ensures further constraints on submitted files, it must be reflected here
+   */
+  private function conjureSubmittedFiles(RuntimeEnvironment $environment) {
+    $extension = current($environment->getExtensionsList());
+    return ["main.{$extension}"];
+  }
+
+  public function check(Exercise $exercise) {
+    try {
+      $config = $this->loader->loadExerciseConfig($exercise->getExerciseConfig()->getParsedConfig());
+      $this->validator->validateExerciseConfig($exercise, $config);
+    } catch (ExerciseConfigException $exception) {
+      $exercise->setBroken(sprintf("Global exercise configuration is invalid: %s", $exception->getMessage()));
+      return;
+    }
+
+    /** @var RuntimeEnvironment $environment */
```

Comment:
After a year in php and PhpStorm... I am not really sure if I like this or not and if it is really necessary :-)

Reply:
Well, I am not fond of it either.

Reply:
I'll gladly remove it when PHP starts supporting type declarations.

Reply:
I know, I know... it wasn't meant to be complaint :-)